File: Nootbook/app/src/main/java/com/example/nootbook/Loginactivity.java
package com.example.nootbook;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class Loginactivity extends AppCompatActivity {

    private FirebaseAuth auth;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_loginactivity);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        TextView switch_btn = findViewById(R.id.switch_to_sign_up);
        switch_btn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                            Intent intent = new Intent(Loginactivity.this, Signup_activity.class);
                            startActivity(intent);
                            finish();
                }
            });

        auth = FirebaseAuth.getInstance();
       Button signinbtn = findViewById(R.id.sign_in_btn);
        signinbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EditText email_input = findViewById(R.id.login_email_input);
                String email = email_input.getText().toString();
                EditText password_input = findViewById(R.id.login_password_input);
                String password =  password_input.getText().toString();
                    signinbtn.setEnabled(false);
                    signinbtn.setText(R.string.processing);
                ProgressBar loginloader = findViewById(R.id.login_loader);
                loginloader.setVisibility(View.VISIBLE);
                auth.signInWithEmailAndPassword(email,password).addOnCompleteListener(Loginactivity.this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        loginloader.setVisibility(View.GONE);
                        signinbtn.setEnabled(true);
                        signinbtn.setText(R.string.sign_in);
                        if (task.isSuccessful()){
                          Intent notintent = new Intent(Loginactivity.this, Noteactivty.class);
                          startActivity(notintent);
                          finish();
                        } else{
                            Toast.makeText(Loginactivity.this, "Invalid Email Id or Password .", Toast.LENGTH_LONG).show();
                        }
                    }
                });
            }
        });


        FirebaseUser session = auth.getCurrentUser();
        if (session!=null){
            Intent intent = new Intent(Loginactivity.this,Noteactivty.class);
            startActivity(intent);
            finish();
        }

    }
}

File: Margmitra/app/src/main/java/com/example/margmitra/LoginActivity.kt
package com.example.margmitra

import android.app.DatePickerDialog
import android.content.Intent
import android.icu.util.Calendar
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.RadioGroup
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_login)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val etDate = findViewById<EditText>(R.id.etDate)

        etDate.setOnClickListener {
            val calendar = Calendar.getInstance()
            val year = calendar.get(Calendar.YEAR)
            val month = calendar.get(Calendar.MONTH)
            val day = calendar.get(Calendar.DAY_OF_MONTH)

            val datePicker = DatePickerDialog(
                this,
                { _, selectedYear, selectedMonth, selectedDay ->
                    etDate.setText("$selectedDay/${selectedMonth + 1}/$selectedYear")
                },
                year, month, day
            )
            datePicker.show()
        }




        var btnRegister = findViewById<Button>(R.id.btnRegister)
        btnRegister.setOnClickListener {
            val rgGender = findViewById<RadioGroup>(R.id.rgGender)
            val selectedId = rgGender.checkedRadioButtonId

            val selectedGender = when (selectedId) {
                R.id.rbMale -> "Male"
                R.id.rbFemale -> "Female"
                else -> ""
            }

            if (selectedGender.isNotEmpty()) {
                Toast.makeText(this, "Selected Gender: $selectedGender", Toast.LENGTH_SHORT).show()
                var int = Intent(this, VerifyOtp::class.java)
                startActivity(int);
            } else {
                Toast.makeText(this, "Please select Male or Female", Toast.LENGTH_SHORT).show()
            }

        }
    }
}